lib/src/Pos/Generator/BlockEvent.hs

```diff
@@ -190,7 +190,10 @@ genBlocksInTree secrets bootStakeholders blockchainTree = do
             }
     -- Partial pattern-matching is safe because we specify
     -- blockCount = 1 in the generation parameters.
-    OldestFirst [block] <- genBlocks blockGenParams
+    -- FIXME avieth what if it's 'Nothing'? Then the list is empty.
```

Comment:
Look at the comment above, it explains why pattern-matching is safe.

Reply:
`genBlock` can give `Nothing` (haven't looked into why).

Reply:
Ah, I see, yes, it was changed to `Nothing` after this code and comment were written.

Reply:
If there's no secret for the leader of a given slot then `Nothing` is given. Maybe that should be an error and the whole thing should fail?